src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,43 @@
 package core.basesyntax;
 
-public class SalaryInfo {
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class SalaryInfo extends Exception {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        int earnedsSalary = 0;
+        StringBuilder report = new StringBuilder();
+        String newLine = System.lineSeparator();
+        report.append("Report for period ")
+                .append(dateFrom).append(" - ")
+                .append(dateTo);
+        for (String name : names) {
+            for (int i = 0; i < data.length; i++) {
+                String[] spBf = data[i].split(" ");
+                if (checkPeriod(dateFrom, dateTo, spBf[0])) {
```

Comment:
Instead of using an array (spBf), consider using more descriptive variable names for the elements to improve code readability.